subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -157,42 +157,52 @@ void add_ipv6_addr_to_zephyr(struct openthread_context *context)
 	}
 }
 
-void add_ipv6_addr_to_ot(struct openthread_context *context)
+void add_ipv6_addr_to_ot(struct openthread_context *context,
+			 const struct in6_addr* addr6)
```

Comment:
`addr6` isn't checked for `NULL` inside this function, which feels dangerous as it is exposed in the header.

Reply:
It's a private header though ,it's not exposed to the application. The pointer is verified before function call, I'm not sure if we should repeat here (we don't verify `context` either).